### PR TITLE
[FLINK-19992][hive] Integrate new orc to Hive source

### PR DESCRIPTION
**Pull request by JingsongLi**
## What is the purpose of the change

After introducing `OrcColumnarRowFileInputFormat`
We need integrate it to Hive, including Hive 2+ and Hive 1.X

## Brief change log

- Introduce `OrcNoHiveColumnarRowInputFormat`
- Integrate orc in `HiveBulkFormatAdapter`

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveBulkFormatAdapter.java [hunk 1]
```diff
@@ -107,24 +114,78 @@ public boolean isSplittable() {
 		return InternalTypeInfo.of(producedRowType);
 	}
 
+	private RowType tableRowType() {
+		LogicalType[] types = Arrays.stream(fieldTypes).map(DataType::getLogicalType).toArray(LogicalType[]::new);
+		return RowType.of(types, fieldNames);
+	}
+
+	private int[] projectedFields() {
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveBulkFormatAdapter.java:122]: We already have a `computeSelectedFields` method, can it be reused?

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveBulkFormatAdapter.java [hunk 2]
```diff
@@ -107,24 +114,78 @@ public boolean isSplittable() {
...
+	private OrcColumnarRowFileInputFormat<?, HiveSourceSplit> createOrcFormat() {
+		return hiveVersion.startsWith("1.") ?
+				OrcColumnarRowFileInputFormat.createPartitionedFormat(
+					OrcShim.createShim(hiveVersion),
+					jobConfWrapper.conf(),
+					tableRowType(),
+					partitionKeys,
+					PARTITION_FIELD_EXTRACTOR,
+					projectedFields(),
+					new ArrayList<>(),
```
Comment (lirui-apache) [inline flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveBulkFormatAdapter.java:167]: `Collections.emptyList()`?

**Review by lirui-apache — APPROVED** [review]
LGTM, thanks for updating

**Comment by JingsongLi** [comment]
E2E failed by `test_streaming_file_sink` looks like network reason which is unrelated to this PR.

**Pull request merged**